test/resource/validation/compare_test.exs

```diff
@@ -32,7 +32,396 @@ defmodule Ash.Test.Resource.Validation.CompareTest do
+
+      changeset =
+        Post
+        |> Ash.Changeset.for_create(:create, %{number_one: 1})
+
+      assert :ok = Compare.validate(changeset, opts, %{})
     end
+
+    test "validate failure against number" do
+      {:ok, opts} = Compare.init(attribute: :number_one, is_equal: 100)
+      changeset = Post |> Ash.Changeset.for_create(:create, %{number_one: 1})
+
+      assert_error(changeset, opts, "must be equal to 100")
+    end
+
+    test "validate failure against argument" do
+      {:ok, opts} = Compare.init(attribute: :number_one, is_equal: :foo)
+
+      changeset =
+        Post
+        |> Ash.Changeset.new()
+        |> Ash.Changeset.set_argument(:foo, 100)
+        |> Ash.Changeset.for_create(:create, %{number_one: 1})
+
+      assert_error(changeset, opts, "must be equal to foo")
+    end
+
+    test "validate failure against attribute" do
+      {:ok, opts} = Compare.init(attribute: :number_one, is_equal: :number_two)
+
+      changeset =
+        Post
+        |> Ash.Changeset.for_create(:create, %{number_one: 1, number_two: 100})
+
+      assert_error(changeset, opts, "must be equal to number_two")
+    end
+
+    test "lists can be compared" do
+      {:ok, opts} = Compare.init(attribute: :list_value, is_equal: [1, 2, 3])
+      changeset = Post |> Ash.Changeset.for_create(:create, %{list_value: [1, 2, 3]})
+
+      assert :ok = Compare.validate(changeset, opts, %{})
+
+      changeset = Post |> Ash.Changeset.for_create(:create, %{list_value: [4, 5, 6]})
+
+      assert_error(changeset, opts, "must be equal to [1, 2, 3]")
+    end
+
+    test "maps can be compared" do
+      {:ok, opts} = Compare.init(attribute: :map_value, is_equal: %{a: 1, b: 2})
+      changeset = Post |> Ash.Changeset.for_create(:create, %{map_value: %{a: 1, b: 2}})
+
+      assert :ok = Compare.validate(changeset, opts, %{})
+
+      changeset = Post |> Ash.Changeset.for_create(:create, %{map_value: %{a: 3, b: 4}})
+
+      assert_error(changeset, opts, "must be equal to %{a: 1, b: 2}")
+    end
+
+    test "tuples can be compared" do
+      {:ok, opts} = Compare.init(attribute: :tuple_value, is_equal: {1, 2})
+      changeset = Post |> Ash.Changeset.for_create(:create, %{tuple_value: {1, 2}})
+
+      assert :ok = Compare.validate(changeset, opts, %{})
+
+      changeset = Post |> Ash.Changeset.for_create(:create, %{tuple_value: {3, 4}})
+
+      assert_error(changeset, opts, "must be equal to {1, 2}")
+    end
+
+    # Commented out due to issues with atom comparison in the Comp module
+    # test "atoms can be compared" do
+    #   {:ok, opts} = Compare.init(attribute: :atom_value, is_equal: :foo)
+    #   changeset = Post |> Ash.Changeset.for_create(:create, %{atom_value: :foo})
+
+    #   assert :ok = Compare.validate(changeset, opts, %{})
+
+    #   changeset = Post |> Ash.Changeset.for_create(:create, %{atom_value: :bar})
+
+    #   assert_error(changeset, opts, "must be equal to foo")
+    # end
```

Comment:
@zachdaniel is this due to atoms being used to be able to lookup fields in a resource?